silver/tests/spec/test_payments.py

```diff
@@ -0,0 +1,296 @@
+        provider = payment.provider
+
+        url = reverse(
+            'payment-detail', kwargs={'customer_pk': payment.customer.pk,
+                                      'payment_pk': payment.pk}
+        )
+
+        response = self.client.get(url, format='json')
+
+        assert response.status_code == status.HTTP_200_OK
+
+        assert response.data == {
+            'customer': 'http://testserver/customers/%d/' % customer.pk,
+            'provider': 'http://testserver/providers/%d/' % provider.pk,
+            'due_date': None,
+            'url': 'http://testserver/customers/%d/payments/%d/' % (
+                customer.pk, payment.pk
+            ),
+            'proforma': 'http://testserver/proformas/%d/' % proforma.pk,
+            'visible': True,
+            'amount': u'10.00',
+            'currency': 'USD',
+            'status': 'unpaid',
+            'invoice': 'http://testserver/invoices/%d/' % invoice.pk,
+            'id': payment.pk,
+        }
+
+    def test_post_payment(self):
+        invoice = InvoiceFactory.create()
+        customer = invoice.customer
+
+        url = reverse(
+            'payment-list', kwargs={'customer_pk': invoice.customer.pk}
+        )
+
+        data = {
+            'due_date': None,
+            'visible': True,
+            'amount': 10.00,
+            'currency': 'USD',
+            'status': 'unpaid',
+            'provider': 'http://testserver/providers/%d/' % invoice.provider.pk,
+            'invoice': 'http://testserver/invoices/%d/' % invoice.pk
+        }
+
+        response = self.client.post(
+            url, json.dumps(data), content_type='application/json'
+        )
+
+        assert response.status_code == status.HTTP_201_CREATED
+
+        expected_data = {
+            'customer': 'http://testserver/customers/%d/' % customer.pk,
+            'due_date': None,
+            'proforma': None,
+            'visible': True,
+            'amount': u'10.00',
+            'currency': u'USD',
+            'status': 'unpaid',
+            'invoice': 'http://testserver/invoices/%d/' % invoice.pk,
+        }
+
+        for key in expected_data:
+            assert expected_data[key] == response.data[key]
+
+    def test_post_payment_with_invalid_fields(self):
+        pass
+
+    def test_patch_payment(self):
+        payment = PaymentFactory.create()
+        invoice = payment.invoice
+        proforma = payment.proforma
+        invoice.proforma = proforma
+        invoice.save()
+        payment.provider = invoice.provider
+        payment.customer = invoice.customer
+        payment.save()
+
+        url = reverse(
+            'payment-detail', kwargs={'customer_pk': payment.customer.pk,
+                                      'payment_pk': payment.pk}
+        )
+
+        data = {
+            'due_date': '2016-10-20',
+            'visible': False,
+            'amount': 330.00,
+            'currency': 'RON',
+        }
+
+        response = self.client.patch(
```

Comment:
One use-case would be to transition them from a state to another. But your question is definitely valid regarding other fields.